clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -76,54 +88,78 @@ public short version() {
     /**
      * Factory method for getting a request object based on ApiKey ID and a buffer
      */
-    public static AbstractRequest getRequest(int requestId, short versionId, ByteBuffer buffer) {
+    public static RequestAndSize getRequest(int requestId, short version, ByteBuffer buffer) {
```

Comment:
I don't think that works because we often don't have (and don't need) a size. We had discussed renaming `RequestAndSize` to `RequestMetadata` and use that class to keep the request version when we remove the builders. However, the request version is already in the header field, so it seems like we don't need it in this other class. So, I left it as `RequestAndSize` for now.